sda/conode.go

```diff
@@ -101,3 +97,26 @@ func (c *Conode) Address() network.Address {
 func (c *Conode) GetService(name string) Service {
 	return c.serviceManager.Service(name)
 }
+
+// ProtocolRegisterName will sign up a new protocol to this Conode.
+// It returns the ID of the protocol.
+func (c *Conode) ProtocolRegisterName(name string, protocol NewProtocol) ProtocolID {
+	u := ProtocolNameToID(name)
+	if n, exists := c.protocolNames[u]; exists {
+		log.Warn("Protocol", n, "already exists - not overwriting")
+		return u
+	}
+	c.protocolNames[u] = name
+	c.protocols[u] = protocol
+	log.Lvl4("Registered", name, "to", u)
+	return u
+}
+
+// ProtocolInstantiate instantiate a protocol from its ID
+func (c *Conode) ProtocolInstantiate(protoID ProtocolID, tni *TreeNodeInstance) (ProtocolInstance, error) {
```

Comment:
If the regular top lvl `ProtocolInstantiate` is not used anymore, I'd remove it - the `sda.ProtocolInstantiate`.